integration-tests/relayderp_test.go

```diff
@@ -0,0 +1,347 @@
+//go:build integration
+
+package integration_tests
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+)
+
+// Testing various connectivity scenarios that requires DERP relay server.
+// These test mimic both public and on-boarded DERP relays.
+
+// TestConnectivityFailureWithoutDerpRelay validates the scenario where the
+// peers behind symmetric NATs are not able to connect to each other if derp
+// relay (public or on-boarded) is not available.
+func TestConnectivityFailureWithoutDerpRelay(t *testing.T) {
+	helper := NewHelper(t)
+	require := helper.require
+	ctx, cancel := context.WithTimeout(context.Background(), 90*time.Second)
+	defer cancel()
+
+	password := "floofykittens"
+	username, cleanup := helper.createNewUser(ctx, password)
+	defer cleanup()
+
+	// Make sure that the nodes don't reach to the public derp relay
+	// that is running at relay.nexodus.io
+	err := os.Setenv("NEX_DERP_RELAY_IP", "127.0.0.1")
```

Comment:
We shouldn't set env vars in integration tests, because if we're running tests in parallel, this could cause failures when tests step on each other. I'm not really sure why the env var is needed here on the host where the test is running. Does it get inherited automatically into each test container or something?

Reply:
Internally we assume that the default public relay that nexodus will use is reachable at relay.nexodus.io, and currently the relay is running and reachable at that instance. As we discuss few weeks back, we don't want to rely on public instance of the relay for integration test, so when the test that checks the default relay test, it tries to reach the public derp relay, so prevent that dns to not resolve to the public ip address, we internally direct it to 127.0.0.1.  So this environment variable is always set to this ip address.

The other alternative is update CreateNode() method or open another interface to use it. I took environment variable approach, because it doesn't change and help me skip updating all CreateNode() calls. And out integration test didn't complain, so didn't revisit it :). But i can work on fixing this. If you have any suggestions that is better than what i mentioned, please let me know i can work on that.

Reply:
Using an env var in general is fine, but I don't think `nexd` is actually seeing the env var with how it's implemented right now.

Reply:
This variable is not for nexd, it's just for the integration tests only, nexd is not using this env variable in any ways. It's only used for here https://github.com/nexodus-io/nexodus/blob/main/integration-tests/helper_test.go#L144

Reply:
@vishnoianil apologies for the misunderstanding.

I would change it to be an argument to CreateNode() if that's the only place it's used. Right now it still isn't safe when running more than one test in parallel that uses that variable.